src/main/scala/query/Assignment.scala

```diff
@@ -0,0 +1,53 @@
+package com.originate.scalypher
+
+import com.originate.scalypher.path.Node
+import com.originate.scalypher.types.Referenceable
+import com.originate.scalypher.types.ReferenceableMap
+import com.originate.scalypher.where.Reference
+import com.originate.scalypher.where.ReferenceWithProperty
+import com.originate.scalypher.where.ObjectReference
+import com.originate.scalypher.where.ValueReference
+
+sealed trait Assignment extends ToQueryWithIdentifiers {
+  def referenceables: Set[Referenceable]
+}
+
+case class RemovePropertyAssignment(reference: ReferenceWithProperty) extends Assignment {
+  def referenceables: Set[Referenceable] =
+    reference.getReferenceable.toSet
+
+  def toQuery(referenceableMap: ReferenceableMap): String =
+    s"${reference.toQuery(referenceableMap)} = NULL"
+}
+
+trait MergeableProperties { self: Referenceable =>
```

Comment:
I think this trait should get extracted to another file, it took me a little while to realize this does't have anything inherently tying it to `Assignment` other than a function returning a `T <: Assignment`